src/main/java/org/brapi/test/BrAPITestServer/service/germ/PedigreeService.java

```diff
@@ -176,22 +172,74 @@ public Optional<PedigreeNodeEntity> getPedigreeNode(String germplasmDbId) {
 		return node;
 	}
 
-	public PedigreeNodeEntity findOrCreatePedigreeNode(String germplasmDbId) throws BrAPIServerException {
-		Optional<PedigreeNodeEntity> nodeOpt = getPedigreeNode(germplasmDbId);
-		PedigreeNodeEntity node;
-		if (nodeOpt.isPresent()) {
-			node = nodeOpt.get();
-		} else {
-			GermplasmEntity germplasm = germplasmService.getGermplasmEntity(germplasmDbId);
+	public List<PedigreeNodeEntity> getPedigreeNodes(List<String> germplasmDbIds) {
+		List<PedigreeNodeEntity> nodes = new ArrayList<>();
+
+		// TODO: Might have to make a custom query for this that fetches the germ eagerly, bc need to compare the germIds.  Have to see if this is a significant performance hit.
+		List<PedigreeNodeEntity> dbNodeList = pedigreeRepository.findByGermplasm_IdIn(germplasmDbIds);
+
+		Map<String, List<PedigreeNodeEntity>> nodesGroupedByGerm = dbNodeList.stream().collect(Collectors.groupingBy(pn -> pn.getGermplasm().getId()));
+
+		nodesGroupedByGerm.forEach((germId, nodesByGerm) -> {
+			if (nodesByGerm.size() > 1) {
+				log.error("multiple pedigree nodes found for a single germplasm");
+			}
+
+			Optional<PedigreeNodeEntity> node = nodesByGerm.stream().findFirst();
+
+			node.ifPresent(nodes::add);
+		});
+
+		return nodes;
+	}
```

Comment:
Can we add a unique constraint on the `pedigree_node` table for the `germplasm_id` column and simplify this method?
```suggestion
	public List<PedigreeNodeEntity> getPedigreeNodes(List<String> germplasmDbIds) {
		return pedigreeRepository.findByGermplasm_IdIn(germplasmDbIds);
	}
```

Reply:
I think probably, but this is a question for @BrapiCoordinatorSelby .  For what it's worth, the current lookup isn't all that inefficient.